### PR TITLE
feat(recorder): standalone gpt-trader record entrypoint with a real recorder identity

### PR DESCRIPTION
**Pull request by Solders-Girdles**
## Summary

PR 3 of #1158 (recorder extraction, five-role composition — M3 sequencing). This lands the last open acceptance criterion: **recording runs standalone via its own entrypoint and keeps collecting while execution is halted/paused.**

- **`features/recorder/market_data_recorder.py`** — `MarketDataRecorder`: polls read-only tickers on an injected cadence (batch `get_tickers` with per-symbol `get_ticker` fallback, mirroring the engine's fetch semantics) and persists price ticks through `PriceTickStore`. Poll failures are logged and never end the run; `stop()` is signal-safe. Nothing in the loop decides a recurrence beyond sleeping the injected interval, per the frequency-headroom directive.
- **`gpt-trader record`** — new CLI command: builds the profile's `ApplicationContainer` (MOCK_BROKER honored), constructs a recorder-owned tick store, runs until SIGINT/SIGTERM. `--once` records a single poll and exits non-zero if nothing was recorded, for verification/scheduling probes. Read-only by construction: the only broker calls are ticker reads.
- **Recorder identity** — `derive_recorder_bot_id` stamps `recorder:<profile>` into recorded events, addressing the #1158 design note about `bot_id=""`. The rehydration semantics for mixed empty/non-empty `bot_id` events are decided deliberately and pinned by a test: `PriceTickStore.rehydrate` filters by **symbol only**, so an engine restart picks up ticks the standalone recorder collected while execution was down (that is the point of the role), and legacy empty-`bot_id` events keep restoring.
- **`cli/services.py`** — `instantiate_container` extracted from `instantiate_bot` (identical validation, fingerprint persistence, and global registration; `instantiate_bot` now delegates) so the record entrypoint composes container services without creating a `TradingBot`.
- **`docs/architecture/ENTRYPOINTS.md`** gains the recorder section.

With this, all #1158 acceptance criteria are met (injected tick state — PR #1159; single snapshot source — PR #1162; standalone recording — here). The engine's WS telemetry stream deliberately stays engine-side until strategy→proposer parity, per docs/decisions/adopt-five-role-composition.md ("the live engine's direct decide→submit path retires only after proposer parity").

## Verification

- `uv run pytest tests/unit -n auto -q` — 6335 passed, 10 skipped (new: 6 recorder-loop tests, identity tests, mixed-bot_id rehydration pin)
- `make ci-required` — green end-to-end (lint, docs audit, typecheck, artifact verify, guardrails, strict-container suite, stage1 smoke)
- `MOCK_BROKER=1 gpt-trader record --profile dev --once` — exit 0; `price_tick` events persisted to `runtime_data/dev/events.db` with `bot_id=recorder:dev` (verified by direct sqlite query)
- `MOCK_BROKER=1 gpt-trader record --profile dev --interval 1` + SIGINT after ~4s — polls on cadence, graceful "recording stopped", exit 0; 14 ticks persisted across runs
- `MOCK_BROKER=1 gpt-trader run --profile dev --dev-fast` — engine unchanged, clean cycle + shutdown
- ruff / black / mypy / agent-naming — clean

Part of #1158.

## Summary by CodeRabbit

* **New Features**
  * Added a standalone `record` CLI command to capture market data without starting trading.
  * Supports continuous recording or `--once`, with selectable profile, optional symbols, and polling interval.
  * Persists captured price tick events under a dedicated recorder identity.
  * Updated architecture documentation with the canonical `record` runtime entrypoint.
* **Bug Fixes**
  * Improved polling resilience (continues after transient failures) with batch-to-per-symbol fallback.
* **Refactor**
  * Centralized container instantiation for consistent bot startup.
* **Tests**
  * Added unit coverage for polling, fetching modes, validation, stop behavior, and recorder identity/rehydration edge cases.

### CODE REVIEW

File: src/gpt_trader/cli/commands/record.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    parser.set_defaults(handler=execute)
+
+
+def execute(args: Namespace) -> int:
+    try:
+        config = services.build_config_from_args(
+            args,
+            include={"symbols", "interval"},
+        )
+        container = services.instantiate_container(config)
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/cli/commands/record.py:62]: Fixed in e26d383b: `execute` now wraps the post-instantiation body in `try/finally` with `clear_application_container()`, mirroring `run.py`. Verified: `--once` still exits 0 with MOCK_BROKER, unit suite green.

File: src/gpt_trader/cli/commands/record.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        bot_id=bot_id,
+    )
+    recorder = MarketDataRecorder(
+        broker=broker,
+        tick_store=tick_store,
+        config=MarketDataRecorderConfig(
+            symbols=tuple(config.symbols),
+            interval_seconds=float(config.interval),
+        ),
+    )
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/cli/commands/record.py]: Fixed in de08eb39 — interval validated on the CLI error path, --once poll guarded with log + exit 1, and non-finite prices rejected in _extract_price (pinned by test_record_once_skips_non_finite_prices).

File: src/gpt_trader/cli/commands/record.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    if args.once:
+        recorded = asyncio.run(recorder.record_once())
+        logger.info(
+            "Recorded one poll",
+            recorded_ticks=recorded,
+            symbols=list(config.symbols),
+            bot_id=bot_id,
+            operation="record",
+        )
+        return 0 if recorded > 0 else 1
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/cli/commands/record.py]: Fixed in de08eb39 — interval validated on the CLI error path, --once poll guarded with log + exit 1, and non-finite prices rejected in _extract_price (pinned by test_record_once_skips_non_finite_prices).

File: src/gpt_trader/features/recorder/market_data_recorder.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+    if not ticker:
+        return None
+    raw_price = ticker.get("price")
+    if raw_price is None:
+        return None
+    try:
+        price = Decimal(str(raw_price))
+    except (DecimalException, ValueError):
+        return None
+    return price if price > 0 else None
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/recorder/market_data_recorder.py:179]: Fixed in de08eb39 — interval validated on the CLI error path, --once poll guarded with log + exit 1, and non-finite prices rejected in _extract_price (pinned by test_record_once_skips_non_finite_prices).

File: var/agents/logging/event_catalog.json [hunk 5]
```diff
@@ -1462,6 +1463,29 @@
...
+        "symbol",
+        "symbols"
+      ],
+      "components": null,
+      "levels": [
+        "ERROR",
+        "INFO",
+        "WARNING"
+      ],
+      "occurrence_count": 8,
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply var/agents/logging/event_catalog.json]: Regenerated in fb562a03 — event catalog now carries the record occurrence count and interval field; `uv run agent-regenerate --verify` is clean on this head.
